### PR TITLE
[AMDGPU] CodeGen for GFX12 VBUFFER instructions

### PR DESCRIPTION
**Pull request by mbrkusanin**

### CODE REVIEW

**Comment by piotrAMD** [comment]
We may need to also update the following files with gfx12 run lines (can be a separate patch):
* test/CodeGen/AMDGPU/llvm.amdgcn.raw.tbuffer.load.d16.ll
* test/CodeGen/AMDGPU/llvm.amdgcn.struct.buffer.load.format.v3f16.ll
* test/CodeGen/AMDGPU/llvm.amdgcn.struct.tbuffer.load.d16.ll

**Review by piotrAMD — APPROVED** [review]
LGTM

**Comment by mbrkusanin** [comment]
Rebased

File: llvm/test/CodeGen/AMDGPU/GlobalISel/extractelement-stack-lower.ll [hunk 1]
```diff
@@ -1,10 +1,105 @@
...
+; GFX12-NEXT:    s_delay_alu instid0(VALU_DEP_1) | instskip(NEXT) | instid1(VALU_DEP_1)
+; GFX12-NEXT:    v_lshlrev_b32_e32 v2, 2, v2
+; GFX12-NEXT:    v_ashrrev_i32_e32 v3, 31, v2
+; GFX12-NEXT:    v_add_co_u32 v0, vcc_lo, v0, v2
+; GFX12-NEXT:    s_delay_alu instid0(VALU_DEP_2)
+; GFX12-NEXT:    v_add_co_ci_u32_e32 v1, vcc_lo, v1, v3, vcc_lo
+; GFX12-NEXT:    global_load_b32 v0, v[0:1], off
+; GFX12-NEXT:    s_waitcnt vmcnt(0)
+; GFX12-NEXT:    s_setpc_b64 s[30:31]
+; GCN-GFX12-LABEL: v_extract_v64i32_varidx:
```
Comment (jayfoad) [inline llvm/test/CodeGen/AMDGPU/GlobalISel/extractelement-stack-lower.ll:34]: These prefixes look wrong. I removed them in 26ff659c39a6.

**Pull request merged**